### PR TITLE
[Merged by Bors] - TO-3294 instrument user api

### PR DESCRIPTION
**Pull request by finiteprods**
**Summary**

adds instrumentation to the main web service so requests can be traced through the service.

### CODE REVIEW

File: discovery_engine_core/web-api/src/handlers.rs [hunk 1]
```diff
@@ -21,15 +22,18 @@ use crate::{
...
-        .map_err(handle_user_state_op_error)?;
+    let user_interests = db.user_state.fetch(&user_id).await.map_err(|err| {
+        error!(
+            "Error fetching interests for user '{}': {:#?}",
+            user_id, err
+        );
+        handle_user_state_op_error(err)
+    })?;
 
     let mut documents = db.documents.clone();
```
Comment (biernacki) [inline discovery_engine_core/web-api/src/handlers.rs:39]: are we interested also how much time ranking takes? @acrrd

File: discovery_engine_core/web-api/src/handlers.rs [hunk 2]
```diff
@@ -52,7 +57,13 @@ pub(crate) async fn handle_user_interaction(
...
+            .map_err(|err| {
+                error!(
+                    "Error updating cois for user '{}' with document '{}': {:#?}",
+                    user_id, body.document_id, err
+                );
+                handle_user_state_op_error(err)
+            })?;
 
         Ok(StatusCode::OK)
     } else {
```
Comment (biernacki) [inline discovery_engine_core/web-api/src/handlers.rs:69]: maybe a `debug` in else block with the `user_id` and `body.document_id` that was not found if that doesn't show up already in log output from `warp::trace::request()`

**Review by biernacki — COMMENTED** [review]
lgtm, just a couple of questions

**Review by biernacki — APPROVED** [review]
lgtm, let me know if you'll need another review depending on @acrrd response

File: discovery_engine_core/web-api/src/handlers.rs (hunk 1 above)
Comment (acrrd) [reply discovery_engine_core/web-api/src/handlers.rs:39]: In the whole request. We can do it here but maybe we can do it in a more comprehensive way from the load balancer.
we can avoid it for a moment and understand later what to do.

**Comment by finiteprods** [comment]
bors merge

**Comment by finiteprods** [comment]
bors merge

**Comment by finiteprods** [comment]
bors merge

**Pull request closed without merging**